tools/bumper/parser_test.go

```diff
@@ -0,0 +1,185 @@
+	}
+
+	AfterEach(func() {
+		By("Removing the temporary file")
+		err := os.Remove(tmpfile.Name())
+		Expect(err).ToNot(HaveOccurred(), "Should successfully remove the temp file")
+	})
+
+	var componentsDummyYaml = []byte(`
+components:
+ multus:
+    url: "https://github.com/intel/multus-cni"
+    commit: "bfaf22964b51b93b08dcb1f10cbd8f8cb9a3603a"
+    metadata:
+      tag: "v3.4.1"
+ linux-bridge:
+    url: "https://github.com/containernetworking/plugins"
+    commit: "ad10b6fa91aacd720f1f9ab94341a97a82a24965"
+    metadata:
+      tag: "v0.8.6"
+ bridge-marker:
+    url: "https://github.com/kubevirt/bridge-marker"
+    commit: "049eb6a796b742767a0be838f6ddb7f79c97dd60"
+    metadata:
+      tag: "0.3.0"
+ kubemacpool:
+    url: "https://github.com/k8snetworkplumbingwg/kubemacpool"
+    commit: "7728339b6c8dcfc9d53cc18ce8e06a910094ed47"
+    metadata:
+      tag: "v0.16.0"
+ nmstate:
+    url: "https://github.com/nmstate/kubernetes-nmstate"
+    commit: "6f41747788235c3363a314dfae46a4ed90e1846a"
+    metadata:
+      tag: "v0.25.0"
+ ovs-cni:
+    url: "https://github.com/kubevirt/ovs-cni"
+    commit: "edb4754d08f49be54c399c938895fe82aab6aa5a"
+    metadata:
+      tag: "v0.12.0"
+ macvtap-cni:
+    url: "https://github.com/kubevirt/macvtap-cni"
+    commit: "14e5a3c7bf516bd4542366a78b7af02f04230ac5"
+    metadata:
+      tag: "v0.2.0"
+`)
+
+	var corruptedDummyYaml = []byte(`
+components:
+ Currepted: "data""
+    Shouldnt': "work"
+`)
+
+	var emptyYaml = []byte(``)
+
+	Describe("ParseComponentsYaml function", func() {
+		Context("when yaml file is empty", func() {
+			BeforeEach(func() {
+				By("Creating a temporary file with empty content")
+				createTempFileWithContent(emptyYaml)
+			})
+			It("should return an error", func() {
+				_, err := parseComponentsYaml(tmpfile.Name())
+				Expect(err).To(HaveOccurred(), "Should Fail to parse file")
+			})
+		})
+		Context("when yaml file is corrupted", func() {
+			BeforeEach(func() {
+				By("Creating a temporary file with corrupted content")
+				createTempFileWithContent(corruptedDummyYaml)
+			})
+			It("should return an error because the unmarshaling failed", func() {
+				By(fmt.Sprintf("Parsing content of temporary file %s", tmpfile.Name()))
+				_, err := parseComponentsYaml(tmpfile.Name())
+				Expect(err).To(HaveOccurred(), "Should Fail to parse file")
+			})
+		})
+		Context("when yaml file is valid", func() {
+			BeforeEach(func() {
+				By("Creating a temporary file with empty content")
+				createTempFileWithContent(componentsDummyYaml)
+			})
+			It("should correctly parse params of all components", func() {
+				By(fmt.Sprintf("Parsing content of temporary file %s", tmpfile.Name()))
+				componentsConfig, err := parseComponentsYaml(tmpfile.Name())
+				Expect(err).ToNot(HaveOccurred(), "Should Succeed to parse file")
+
+				By("Checking map is not empty")
+				Expect(componentsConfig.Components).ToNot(BeEmpty())
+
```

Comment:
Gomega do a DeepEquals so is safe to compare the maps meaning that you can just compare input and output config instead of "unrolling" it, although maybe is not as clear at error we can fix that doing at gomega custom matcher for Component type.

Reply:
Not sure what you mean here.
I don't have anything ready to compare componentsConfig to.
Do you mean I should create a static componentsConfig with the correct data and just do `if componentsConfig == ExpectedcomponentsConfig`?

Reply:
Github is doing tricks, this comments was not done here...

Reply:
Done